Kernel_23/include/CGAL/Kernel/hash_functions.h

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) 2019
+// GeometryFactory (France)
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0+
+// 
+//
+// Author(s)     : Simon Giraudot
+ 
+#ifndef CGAL_KERNEL_HASH_FUNCTIONS_H
+#define CGAL_KERNEL_HASH_FUNCTIONS_H
+
+namespace CGAL
+{
+
+template <typename K>
+inline std::enable_if_t<std::is_floating_point<typename K::FT>::value, std::size_t>
+hash_value (const Aff_transformation_2<K>& transform)
+{
+  std::size_t result = boost::hash_value(transform.cartesian(0,0));
```

Comment:
`hash_value` must be called by ADL, without the namespace. Or, if you want to specify a namespace, it must be `CGAL`.

Travis tests are fully red: https://travis-ci.org/CGAL/cgal/builds/541190492. It probably does not compile on your machine either.

Reply:
It does compile using Clang (I just double-checked), but fails with GCC.

But in that case, `hash_value` is applied on a double/float type, so it does come from Boost, doesn't it? I pretty much copied what I did for [`CGAL::grid_simplify_points()`](https://github.com/CGAL/cgal/blob/master/Point_set_processing_3/include/CGAL/grid_simplify_point_set.h#L80) a few years ago.

Reply:
Actually my suggestion was wrong. The correct one is:
  - call by ADL (without the namespace),
  - but add a `using boost::hash_value;` before, for the cases where the FT does not have a namespace, like `double`.

Maybe the errors in Travis are just about a missing `#include` for `boost::hash_value`. But in general `hash_value` must be called by ADL (with the additional `using`).